analysis/spectra_chisq/_model_spectra.py

```diff
@@ -0,0 +1,261 @@
+"""The ``_model_spectra`` module models spectra by varying the dust extinction from the Milky Way.
+
+The model used is ``sncosmo.CCM89Dust``. The parameters A_v and amplitude are fit to the observed spectra. The best fit  parameters are determined by ``scipy.optimize.curve_fit``
+The fit parameters are saved to a fits file.
+"""
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy.table import Table
+
+from functools import partial
+from scipy.optimize import curve_fit
+
+import sncosmo
+import sndata
+from sndata.csp import DR1, DR3
+
+
+dr1, dr3 = DR1(), DR3()
+dr1.download_module_data()
+dr3.download_module_data()
+dr3.register_filters(force=True)
+
+
+def jd_to_mjd(date):
+    """Convert JD date to MJD
+    
+    Args:
+        date (float): JD date
+        
+    Returns:
+        (float) MJD date
+    """
+    
+    return date + 2400000.0
+
+
+def get_t0(obj_id, tbl=dr3.load_table(3), mjd=False):
+    """Get peak brightness of SN in B-band
+    
+    Args:
+        obj_id (str): specific SN object id
+        tbl (Table): 
+        jd (bool): True if the date in `tbl` is JD. False if it is MJD.
+    
+    Returns:
+        t0 (float): MJD date of max brightness
+    """
+    
+    # Get data for specific SN
+    obj_dat = tbl[tbl['SN'] == obj_id]
+    # Get t0 for specific SN
+    t0 = obj_dat['T(Bmax)']
+    if not mjd:
+        # Convert t0 to MJD date
+        t0_mjd = jd_to_mjd(date=t0)
+    
+    return t0_mjd
+
+
+def create_model(model, obj_id, a_v, redshift, amplitude, photo_data, t0_tbl=dr3.load_table(3),
+                        r_v=3.1, mag_sys='ab'):
+    """Model Milky Way dust extinction. Set parameters: EBV (color excess), t0 (time of max brightness in B band),
+    and redshift.
+    Note: DR1 neglects extinction from host galaxy, so only MW dust is modeled currently
+
+    
+    Note: typical Av ~ 0.42 mag via Av = Rv * E(B-V) = 3.1 * 0.134mag ~ 0.42mag
+    where Milky Way ebv = 0.134 (from Table 8 of DR3/Folatelli et al. 2013)
+    and Milkway Way Rv = 3.1 from DR1 at the end of pg 9
+    
+    Args:
+        model (): sncosmo dust model
+        obj_id (str):
+        a_v (float): visual (V filter) extinction in units of magnitude
+    
+    Returns:
+        model (): Updated sncosmo model
+    """
+    
+    ebv = a_v / r_v # unit: mags?
+    # Get time of peak brightness
+    t0 = get_t0(obj_id=obj_id)
+    model.set(z=redshift, t0=t0, mwebv=ebv, amplitude=amplitude)
+    
+    return model
+
+
+def get_obs_wave_and_flux(obj_id, spectra, date):
+    """ """
+    
+    spectrum = spec_data[spec_data['time'] == date]
```

Comment:
`spec_data` isn't being passed as an argument so it defaults to using the `spec_data` defined in the global scope (i.e., the `spec_data` variable you define at the end of this file). In general, this function can probably also be dropped and replaced with `spec_data[spec_data['time'] == date]`.